sections/exchange/hooks/useExchange.tsx

```diff
@@ -163,10 +154,9 @@ const useExchange = ({
 	const [txApproveModalOpen, setTxApproveModalOpen] = useState<boolean>(false);
 	const setOrders = useSetRecoilState(ordersState);
 	const setHasOrdersNotification = useSetRecoilState(hasOrdersNotificationState);
-	const gasSpeed = useRecoilValue<keyof GasPrices>(gasSpeedState);
-	const customGasPrice = useRecoilValue(customGasPriceState);
 	const { selectPriceCurrencyRate, selectedPriceCurrency } = useSelectedPriceCurrency();
 	const network = useRecoilValue(networkState);
+	const { gasPrice, gasPriceWei, gasPrices, gasConfig } = useGas();
```

Comment:
i see that you use `gasConfig` further down, but what are the remaining `gasPrice, gasPriceWei, gasPrices` used for?

Reply:
so useExchange is a pretty big file - in regards to this one -- useExchange is a large file so the places where some of these values are not obvious from the pr

`gasPrice` is calculated in `useGas` now,
`gasPriceWei` is calculated in `useGas` based on the value of `gasPrice`,
`gasPrices` is the result of `const { useEthGasPriceQuery } = useSynthetixQueries();` which is used in many places around the app - i consolidated it into useGas as `gasPrices`

in useExchange, here are the lines where these values are used:

[line 523-526](https://github.com/Kwenta/kwenta/blob/1fef9f13d8107ae3c2b1a9fe197028e5a836c124/sections/exchange/hooks/useExchange.tsx#L523-L526)
[line 554](https://github.com/Kwenta/kwenta/blob/1fef9f13d8107ae3c2b1a9fe197028e5a836c124/sections/exchange/hooks/useExchange.tsx#L554)
[line 592-618](https://github.com/Kwenta/kwenta/blob/1fef9f13d8107ae3c2b1a9fe197028e5a836c124/sections/exchange/hooks/useExchange.tsx#L592-L618)
[line 697-790](https://github.com/Kwenta/kwenta/blob/1fef9f13d8107ae3c2b1a9fe197028e5a836c124/sections/exchange/hooks/useExchange.tsx#L697-L790)
[line 1099](https://github.com/Kwenta/kwenta/blob/1fef9f13d8107ae3c2b1a9fe197028e5a836c124/sections/exchange/hooks/useExchange.tsx#L1099)